src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/SynthesizedClosureEnvironment.cs

```diff
@@ -89,15 +96,31 @@ private static void AssertIsClosureScopeSyntax(SyntaxNode syntaxOpt)
 
         public override ImmutableArray<Symbol> GetMembers()
         {
-            var members = base.GetMembers();
-            if ((object)StaticConstructor != null)
+            if (_members.IsDefault)
             {
-                members = ImmutableArray.Create<Symbol>(StaticConstructor, SingletonCache).AddRange(members);
+                var builder = _membersBuilder;
+                if ((object)StaticConstructor != null)
+                {
+                    builder.Add(StaticConstructor);
+                    builder.Add(SingletonCache);
+                }
+                builder.AddRange(base.GetMembers());
+                _members = builder.ToImmutableAndFree();
+                _membersBuilder = null;
             }
 
-            return members;
+            return _members;
         }
 
+        /// <summary>
+        /// All fields should have already been added as synthesized members on the
+        /// <see cref="CommonPEModuleBuilder" />, so we don't want to duplicate them here.
+        /// </summary>
+        internal override IEnumerable<FieldSymbol> GetFieldsToEmit()
+            => (object)SingletonCache != null
+            ? SpecializedCollections.SingletonEnumerable(SingletonCache)
```

Comment:
>SpecializedCollections.SingletonEnumerable(SingletonCache) [](start = 14, length = 58)

This implementation makes an assumption that base.GetMembers() doesn't return any fields. It is not clear why this is a safe assumption. We also do not assert this anywhere.